schema/testing.proto

```diff
@@ -238,4 +230,128 @@ message Test {
 
   // A description of the test.
   string description = 3;
+
+  // A blueprint is an explicit definition of methods and requests that are needed
+  // to be made to test this specific test case. Ideally this would be represented
+  // by something more robust like CEL, but as of writing this, I am unsure if CEL
+  // is ready.
+  message Blueprint {
+    // The id of this blueprint.
+    string id = 1;
+
+    // A description of this blueprint.
+    string description = 2;
+
+    // A message representing a method invocation.
+    message Invocation {
+      // The fully qualified name of the showcase method to be invoked.
+      string method = 1;
+
+      // The request to be made if a specific request is necessary.
+      google.protobuf.Any request = 2;
+    }
+
+    // An ordered list of method calls that can be called to trigger this test.
+    repeated Invocation invocations = 3;
+  }
+
+  // The blueprints that will satisfy this test. There may be multiple blueprints
+  // that can signal to the server that this test case is being exercised. Although
+  // multiple blueprints are specified, only a single blueprint needs to be run to
+  // signal that the test case was exercised.
+  repeated Blueprint blueprints = 4;
+}
+
+// An issue found in the test.
+message Issue {
+  // The different potential types of issues.
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+
+    // The test was never instrumented.
+    SKIPPED = 1;
+
+    // The test was started but never confirmed.
+    PENDING = 2;
```

Comment:
How does Showcase know this one?

Reply:
If the initial request in any of the blueprints was called then the test is pending, unless it has been confirmed or failed.